packages/ra-ui-materialui/src/list/datagrid/Datagrid.tsx

```diff
@@ -150,6 +153,23 @@ export const Datagrid: React.ForwardRefExoticComponent<
         total,
     } = useListContextWithProps(props);
 
+    const resourceKeys =
+        React.Children.map(children, field => {
+            if (!isValidElement(field)) {
+                return null;
+            }
+
+            return field.props.source;
+        })?.filter(value => typeof value === 'string') || [];
```

Comment:
I understand that fields without source won't be checked, am I right? This deserves a warning in the doc.

Reply:
Done